selftests/unit/test_utils_process.py

```diff
@@ -321,6 +336,36 @@ def test_run_unicode_output(self):
         self.assertEqual(result.stdout, encoded_text)
         self.assertEqual(result.stdout_text, text)
 
+    def test_run_with_timeout_ugly_cmd(self):
+        content = (REFUSE_TO_DIE)
+        with script.TemporaryScript("refuse_to_die", content) as exe:
+            cmd = "%s '%s'" % (sys.executable, exe.path)
+            # Wait 1s to set the traps
+            res = process.run(cmd, timeout=1, ignore_status=True)
+            self.assertLess(res.duration, 100, "Took longer than expected, "
+                            "process probably not interrupted by Avocado.\n%s"
+                            % res)
+            self.assertNotEqual(res.exit_status, 0, "Command finished without "
+                                "reporting failure but should be killed.\n%s"
+                                % res)
+
+    def test_run_with_negative_timeout_ugly_cmd(self):
```

Comment:
Even though the timeout is quite large, consider using the boiler plate code we've using for other tests that involve timing issues / timeouts:

```
    @unittest.skipIf(int(os.environ.get("AVOCADO_CHECK_LEVEL", 0)) < 2,
                     "Skipping test that take a long time to run, are "
                     "resource intensive or time sensitve")
```